### PR TITLE
OCPBUGS-33149: azure-path-fix: get client secret from k8s secret

### PR DESCRIPTION
**Pull request by flavianmissi**
/assign @dmage @xiuwang

### CODE REVIEW

**Comment by flavianmissi** [comment]
Apparently we're running the `TestS3Minio` on all clouds 🤔 It seems to be perma-failing because of the azure-path-fix job, which is running on e2e-azure-operator because it's detecting that the operator is running on Azure, even though the `TestS3Minio` suite is configuring the storage backend to s3. The weird thing is that this change seems to be introducing the error... I'm going to retest it for now, and tomorrow I'll investigate deactivating the job if the operator is configured with s3.

/retest

**Comment by flavianmissi** [comment]
After closing inspection of the failing job it looks like the operator is degraded for a reason other than the azure-path-fix job failing 🤔 
```
          - lastTransitionTime: "2024-05-02T18:10:23Z"
            message: The deployment has not completed
            reason: DeploymentNotCompleted
            status: "True"
            type: Progressing
          - lastTransitionTime: "2024-05-02T18:12:26Z"
            message: The deployment does not have available replicas
            reason: Unavailable
            status: "True"
            type: Degraded
```
Also seeing these in the operator logs (though I don't quite know how to interpret them):
```
E0502 18:12:32.164869       1 controller.go:377] unable to sync: unable to sync storage configuration: failed to generate shared secrets data: invalid secret for aws credentials, requeuing
```
/retest

**Comment by flavianmissi** [comment]
got a different failure this time, looks like a ci hiccup:
```
{  "e2e-azure-operator" pod "e2e-azure-operator-test" failed: could not watch pod: ci-op-1977i9v8/e2e-azure-operator-test was deleted
Link to step on registry info site: https://steps.ci.openshift.org/reference/test
Link to job on registry info site: https://steps.ci.openshift.org/job?org=openshift&repo=cluster-image-registry-operator&branch=master&test=e2e-azure-operator}
```
/retest

File: pkg/resource/azurepathfixjob.go [hunk 1]
```diff
@@ -99,7 +99,15 @@ func (gapfj *generatorAzurePathFixJob) expected() (runtime.Object, error) {
 		{Name: "AZURE_CONTAINER_NAME", Value: azureStorage.Container},
 		{Name: "AZURE_CLIENT_ID", Value: azureCfg.ClientID},
 		{Name: "AZURE_TENANT_ID", Value: azureCfg.TenantID},
-		{Name: "AZURE_CLIENT_SECRET", Value: azureCfg.ClientSecret},
+		{Name: "AZURE_CLIENT_SECRET", ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				Optional: &optional,
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: defaults.CloudCredentialsName,
```
Comment (dmage) [inline pkg/resource/azurepathfixjob.go:106]: What will happen if Azure is configured via `REGISTRY_STORAGE_AZURE_ACCOUNTKEY` in the user configuration secret?

Comment (flavianmissi) [reply pkg/resource/azurepathfixjob.go:106]: just realized you asked about the user configuration secret - that won't actually work unless the user manually updates the non-user secret, we have a separate bug for that https://issues.redhat.com/browse/OCPBUGS-32703, but this behavior is very old so it's a low priority fix.

**Pull request merged**